### PR TITLE
fix(route): fix Spotify TypeError

### LINKED ISSUE #15648: Spotify TypeError
### Routes

```routes
/spotify/show/:id
```

### Full routes

```fullroutes
/spotify/show/:id
```

### Related documentation

https://docs.rsshub.app/zh/routes/multimedia#show

### What is expected?

should return correct rss content

### What is actually happening?

TypeError: got.post(...).json is not a function
screenshot-20240521-183153

### Deployment information

Self-hosted

### Deployment information (for self-hosted)

_No response_

### Additional info

```shell
I noticed that the official method of sending http requests has been updated： https://docs.rsshub.app/joinus/new-rss/start-code#basic-code, maybe should change to utils/ofetch
```

### This is not a duplicated issue

- [X] I have searched [existing issues](https://github.com/DIYgod/RSSHub/issues) to ensure this bug has not already been reported

### PR DESCRIPTION
**Pull request by ttl97**
## Involved Issue / 该 PR 相关 Issue

Close #15648

## Example for the Proposed Route(s) / 路由地址示例

```routes
/spotify/artist/6k9TBCxyr4bXwZ8Y21Kwn1
/spotify/top/artists
/spotify/playlist/4UBVy1LttvodwivPUuwJk2
/spotify/saved/50
/spotify/top/tracks
/spotify/show/5CfCWKI5pZ28U0uOzXkDHe
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
fix Spotify TypeError by replacing `got` with `ofetch`.

### CODE REVIEW

File: lib/routes/spotify/artist.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { Route } from '@/types';
 import utils from './utils';
-import got from '@/utils/got';
 import { parseDate } from '@/utils/parse-date';
+import { ofetch } from 'ofetch';
```
Comment (TonyRL) [inline lib/routes/spotify/artist.ts:4]:
```suggestion
import ofetch from '@/utils/ofetch';
```

File: lib/routes/spotify/artists-top.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { Route } from '@/types';
 import utils from './utils';
-import got from '@/utils/got';
+import { ofetch } from 'ofetch';
```
Comment (TonyRL) [inline lib/routes/spotify/artists-top.ts:3]:
```suggestion
import ofetch from '@/utils/ofetch';
```

File: lib/routes/spotify/playlist.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { Route } from '@/types';
 import utils from './utils';
-import got from '@/utils/got';
 import { parseDate } from '@/utils/parse-date';
+import { ofetch } from 'ofetch';
```
Comment (TonyRL) [inline lib/routes/spotify/playlist.ts:4]:
```suggestion
import ofetch from '@/utils/ofetch';
```

File: lib/routes/spotify/saved.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { Route } from '@/types';
 import utils from './utils';
-import got from '@/utils/got';
 import { parseDate } from '@/utils/parse-date';
+import { ofetch } from 'ofetch';
```
Comment (TonyRL) [inline lib/routes/spotify/saved.ts:4]:
```suggestion
import ofetch from '@/utils/ofetch';
```

File: lib/routes/spotify/tracks-top.ts [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import { Route } from '@/types';
 import utils from './utils';
-import got from '@/utils/got';
+import { ofetch } from 'ofetch';
```
Comment (TonyRL) [inline lib/routes/spotify/tracks-top.ts:3]:
```suggestion
import ofetch from '@/utils/ofetch';
```

Comment (ttl97) [reply lib/routes/spotify/tracks-top.ts]: all fixed.

**Pull request merged**